src/services/search-builder.js

```diff
@@ -30,13 +27,17 @@ function SearchBuilder(model, opts, params, searchFields) {
       const condition = {};
       const searchValue = params.search.replace(/[-[\]{}()*+!<=:?./\\^$|#\s,]/g, '\\$&');
       const searchRegexp = new RegExp(`.*${searchValue}.*`, 'i');
+      const searchNumber = Number(params.search);
 
       if (value.instance === 'ObjectID') {
-        try {
+        if (new RegExp('^[0-9a-fA-F]{24}$').test(params.search)) {
           condition[key] = opts.mongoose.Types.ObjectId(params.search);
           pushCondition(condition, key);
-        } catch (error) {
-          console.error(error);
+        }
+      } else if (value.instance === 'Number') {
+        if (!Number.isNaN(searchNumber)) {
```

Comment:
[Variables] `searchNumber` seems to be used only here. Thus you should declare your variable only in this scope (it avoids var assignment and unnecessary cast)